src/app/oauth/components/oauth.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AppState } from '../../appcommon/services/app.service';
+import { UserInfoService } from '../services/userinfo.service';
+
+@Component({
+  selector: 'app-oauth',
+  templateUrl: './oauth.component.html',
+  styleUrls: ['./oauth.component.scss']
+})
+export class OAuthComponent implements OnInit {
+  userInfo: any;
+  accessToken: string;
+
+  constructor(
+    private appState: AppState,
+    private activeRoute: ActivatedRoute,
+    private router: Router,
+    private service: UserInfoService
+  ) {
+    this.activeRoute.params.subscribe(params => {
+      const url = window.location.href;
+      if (url.split('#').length > 1) {
+        const responseParameters = url.split('#')[1].split('&');
```

Comment:
This type of URL splitting is very shaky.. as # is not guaranteed to be present in the URL.

We should be using [paramMap](https://angular.io/guide/router#activated-route-in-action) to read params.

Reply:
This is while reading the access token from URL. This is standard OAuth specification that the values will be separated by #